docs/tutorials/portable-consent.md

```diff
@@ -0,0 +1,137 @@
+
+<img src={requeststab} style={{width:"375px"}}/>
+
+You can then give your user the option to block or allow the contact.
+
+<img src={allowblock} style={{width:"375px"}}/>
+
+If your user selects **Allow**, display the conversation with the contact in your user’s main inbox view, such as on a **Messages** tab.
+
+<img src={messagestab} style={{width:"375px"}}/>
+
+If your user selects **Block**, remove the conversation with the contact from all views in your user’s inbox. This ensures that your user’s main inbox view is populated with only conversations with contacts they want to message with. Meanwhile, conversations with potential new contacts are accessible in a secondary view, and conversations with contacts they’ve blocked stay out of view completely.
+
+You can also enable your users to block a contact in a previously accepted conversation. Likewise, you can enable your users to allow (unblock) a previously blocked contact.
+
+To ensure that your users’ consent settings are consistently applied in any app built with XMTP, contact consent states are stored and accessible to apps in a consent list on the XMTP network.
+
+For example, if your user blocked a contact on a different device or in another app and then uses your app, you can load the latest consent list to be sure to not mistakenly display conversations with the blocked contact.
+
+## Understand contact consent states
+
+A contact can have one of three consent states:
+
+- Unknown
+- Allowed
+- Blocked
+
+Here are some of the ways contact consent states are set:
+
+**Unknown**
+
+- For a new conversation that a new contact wants to start with your user, the contact consent state is set to `unknown`.
+    
+    For example, you can consider displaying a conversation with an `unknown` contact on a **Requests** tab and give your user the option to block or allow the contact.
+    
+- For a preexisting conversation with only messages from a contact that hasn’t had its consent state updated on the network, the contact consent state is set to `unknown` by default.
+    
+    For example, you can consider displaying a conversation with an `unknown` contact on a **Requests** tab and give your user the option to block or allow the contact.
+    
+**Allowed**
+
+- For a new conversation that your user requests with a contact, the contact consent state is set to `allowed`. Your user’s creation of the conversation with the contact is considered as consenting to the conversation with the contact.
+    
+    For example, you can consider displaying the conversation with the `allowed` contact on the **Messages** tab.
+    
+- For a preexisting conversation with one or more messages from a contact that hasn’t had its consent state updated on the network (`unknown`) and your user has taken the action to send a message in response, update the contact consent state to `allowed`. You can consider the response from your user as consenting to the conversation with the contact.
```

Comment:
I wonder if this needs to more explicitly state that the contact consent state should be updated in the consent list on the network?

Same for the Allow and Block actions below.

Reply:
Well I think there is two scenarios here. One is a preexisting conversations that pre-dates this consent work. I think thats the case being handled here. The other scenario is a new conversation on a device that doesn't have any consent type functionality yet. In that case I think the network should handle it.

Reply:
I'm trying something with the update to account for this... Let's see if it works! ;-) More soon...